go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -263,10 +290,24 @@ func (c *Controller) convertANPPeersToIPs(anp *adminNetworkPolicyState) error {
 }
 
 // convertANPPeersToIPSet creates a set of peerIPs for all the peers passed as argument
-// This function also takes care of populating the adminNetworkPolicyPeer.namespaces cache
+// This function also takes care of populating the adminNetworkPolicyPeer.namespaces, adminNetworkPolicyPeer.nodes cache
 // It also adds up all the peerIPs that are supposed to be present in the created AS and returns them on
 // a per-rule basis so that the actual ops to transact these into the AS can be constructed using that
-func (c *Controller) convertANPPeersToIPSet(peers []*adminNetworkPolicyPeer, peerIPs sets.Set[string]) error {
+// Since we have to loop through all the peer pods here, we also take the opportunity to populate our
+// namedPorts cache in rule.ports so that it can be used to construct ACL matches
+func (c *Controller) convertANPPeersToIPSet(peers []*adminNetworkPolicyPeer, peerIPs sets.Set[string], ports []*libovsdbutil.NetworkPolicyPort) error {
+	// Process NamedPorts if any
+	namedPortRepresentations := map[string]*[]libovsdbutil.NamedNetworkPolicyPort{}
```

Comment:
since I changed the struct to be independent (based on previous comment) this part is gone, no need to copy we can straight use the new list of named ports and it also makes the code cleaner.